botorch/acquisition/multi_objective/monte_carlo.py

```diff
@@ -415,6 +426,17 @@ def __init__(
                 category=BotorchWarning,
             )
             self.sampler.base_samples = None
+        elif self._uses_matheron and self.sampler.batch_range != (0, -1):
+            warnings.warn(
+                message=(
+                    "sampler.batch_range is not (0, -1). qNEHVI requires that the "
+                    "sampler.batch_range is (0, -1) with GPs that use Matheron's rule "
+                    "for sampling, in order to properly collapse batch dimensions. "
+                    "Overwriting the batch_range with (0, -1)!"
+                ),
+                category=BotorchWarning,
+            )
+            self.sampler.batch_range = (0, -1)
```

Comment:
So, this essentially makes SAA work correctly by making `collapse_batch_dims` function properly. If you don't do this,  when `X` is `b x q x d` the sampler base sample shape ends up being `num_samples x b x m'` where `m'` is the proper # of base samples for Matheron. This means, if I take a `q x d`-dim `X` and evaluate `X.repeat(5, 1, 1)`, I will get `5` different acquisition values.

The reason I went for overwriting is that we typically don't set the `batch_range` of the sampler and just use the default `(0, -2)` because that is what you want for every other model. So, I figured we can just correct the default here. Maybe an error is more fool-proof. Let the user fix it themselves